weatherman.py

```diff
@@ -0,0 +1,194 @@
+import argparse
+import calendar
+from colorama import Fore, Style
+import csv
+import glob
+import os
+
+def convert_month(weather_month_input):
+    try:
+        month_number = int(weather_month_input)
+        if 1 <= month_number <= 12:
+            return calendar.month_abbr[month_number]
+    except ValueError:
+        raise ValueError("Invalid input")
+    
+def read_files(weather_year, weather_month):
+    weather_details = []
+
+    base_directory = r"D:\Cogent Labs\weatherfiles\weatherfiles"
+    pattern = f"*{weather_year}*"
+    if weather_month:
+        pattern = f"*{weather_year}*{weather_month}*"
+
+    files_to_process = glob.glob(os.path.join(base_directory, pattern))
```

Comment:
single function has a single responsibility, Please create a separate function for that, in that function only creates weather file names,